pkg/kds/global/components.go

```diff
@@ -70,25 +70,34 @@ func ProvidedFilter(clusterID string, r model.Resource) bool {
 
 func SetupComponent(rt runtime.Runtime) error {
 	syncStore := sync_store.NewResourceSyncer(kdsGlobalLog, rt.ResourceStore())
+	readOnlyResourceManager := rt.ReadOnlyResourceManager()
 
 	clientFactory := func(clusterIP string) client.ClientFactory {
 		return func() (kdsClient client.KDSClient, err error) {
 			return client.New(clusterIP, rt.Config().KDS.Client)
 		}
 	}
 
-	for _, zone := range rt.Config().Mode.Global.Zones {
-		log := kdsGlobalLog.WithValues("clusterIP", zone.Remote.Address)
+	zones := &system.ZoneResourceList{}
+	err := readOnlyResourceManager.List(context.Background(), zones)
+	if err != nil {
+		kdsGlobalLog.Error(err, "unable to list Zone resources")
+		return err
+	}
```

Comment:
As far as I understand this code, any changes in Zone requires CP to restart, because `SetupComponent` is run only when CP starts. Is this something that bi directional KDS will fix?

Reply:
Again, yes - that is the idea. This PR provides the Resource boilerplate for the next-step bi-directional KDS.